modules/dicehub/extension/provider.go

```diff
@@ -63,13 +69,29 @@ func (p *provider) Init(ctx servicehub.Context) error {
 				}),
 			))
 	}
-	go func() {
-		err := p.extensionService.InitExtension(FilePath, false)
-		if err != nil {
-			panic(err)
-		}
-		logrus.Infoln("End init extension")
-	}()
+
+	var once sync.Once
+	p.InitExtensionElection.OnLeader(func(ctx context.Context) {
+		once.Do(func() {
+			var doneValue atomic.Value
+			err := p.extensionService.InitExtension(FilePath, false)
+			if err != nil {
+				panic(err)
+			}
+			doneValue.Store(true)
+			logrus.Infoln("End init extension")
+
+			go func() {
+				select {
+				case <-ctx.Done():
+					done, ok := doneValue.Load().(bool)
+					if !ok || !done {
+						panic(fmt.Errorf("init extension not done but lost leader"))
```

Comment:
No need panic if leader lost.
Another leader will initExtensions.

And your go func should put before line #77.